pkg/resource/replication_group/delta_util.go

```diff
@@ -62,3 +79,74 @@ func engineVersionsMatch(
 
 	return false
 }
+
+// compares desired.Spec to last applied slow log config in latest.Status to determine
+// whether to invoke an update or not. This comes down to only trying to apply a particular configuration
+// once.
+func slowLogRequiresUpdate(desired *resource, latest *resource) bool {
+	var desiredConfig *svcapitypes.LogDeliveryConfigurationRequest
+	var lastRequestedConfig *svcapitypes.LogDeliveryConfiguration
+
+	// retrieve desiredConfig from desired Spec
+	if desired.ko.Spec.LogDeliveryConfigurations != nil {
+		for _, config := range desired.ko.Spec.LogDeliveryConfigurations {
+			if config.LogType != nil && *config.LogType == "slow-log" {
+				desiredConfig = config.DeepCopy()
+			}
+		}
+	}
+
+	// retrieve lastRequestedConfig from latest Status
+	if latest.ko.Status.LastRequestedLogDeliveryConfigurations != nil {
+		for _, config := range latest.ko.Status.LastRequestedLogDeliveryConfigurations {
+			if config.LogType != nil && *config.LogType == "slow-log" {
+				lastRequestedConfig = config.DeepCopy()
+			}
+		}
+	}
+
+	// if both nil, no update to be done.
+	// if desired nil and latest non-nil, then the intention could be to disable; update required.
+	// similarly, if desired non-nil and latest nil, then log delivery must be enabled; update required.
+	// if both non-nil, compare values to determine whether an update is required
+	if desiredConfig == nil && lastRequestedConfig == nil {
+		return false
+	} else if desiredConfig == nil || lastRequestedConfig == nil {
```

Comment:
What if `latestRequestConfig != nil` && `desiredConfig == nil`? The if statement in line 114 will not catch it